cdap-common/src/main/java/co/cask/cdap/common/logging/AbstractLoggingContext.java

```diff
@@ -30,15 +32,38 @@
  */
 public abstract class AbstractLoggingContext implements LoggingContext {
 
-  private static final Function<SystemTag, String> SYSTEM_TAG_TO_STRING = new Function<SystemTag, String>() {
-    @Override
-    public String apply(LoggingContext.SystemTag systemTag) {
-      return systemTag.getValue();
-    }
-  };
+  public static final String TAG_YARN_APP_ID = ".yarnAppId";
+  public static final String TAG_YARN_CONTAINER_ID = ".yarnContainerId";
 
   // Map looks not efficient here, it might be better to use set
-  private final Map<String, SystemTag> systemTags = Maps.newHashMap();
+  private final Map<String, SystemTag> systemTags;
+
+  protected AbstractLoggingContext() {
+    systemTags = Maps.newHashMap();
+
+    // Try picking up the YARN container id from the env, parse it and set it to context
+    String containerId = System.getenv("CONTAINER_ID");
+    if (containerId == null) {
+      return;
+    }
+
+    try {
+      ContainerId yarnContainerId;
+      try {
+        // For Hadoop 2.6+, use ContainerId.fromString(String)
+        Method fromString = ContainerId.class.getMethod("fromString", String.class);
+        yarnContainerId = (ContainerId) fromString.invoke(null, containerId);
+      } catch (NoSuchMethodException e) {
+        // This is for older Hadoop
+        yarnContainerId = ConverterUtils.toContainerId(containerId);
+      }
+
+      setSystemTag(TAG_YARN_APP_ID, yarnContainerId.getApplicationAttemptId().getApplicationId().toString());
+      setSystemTag(TAG_YARN_CONTAINER_ID, yarnContainerId.toString());
+    } catch (Exception e) {
+      // Ignore any exception
```

Comment:
It would be good to log the exception in debug level, so that we can figure out issues when we don't see the container id in the context

Reply:
fixed